shared/src/iosMain/kotlin/com/fbada006/shared/utils/ImagePickerFactory.ios.kt

```diff
@@ -0,0 +1,17 @@
+package com.fbada006.shared.utils
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import platform.UIKit.UIViewController
+
+actual class ImagePickerFactory(
+    private val rootController: UIViewController
+){
+
+    @Composable
```

Comment:
Using the same pattern as you have in android, you can create the iOS one. Instead of passing the `UIViewController` in the constructor of this class, you can get it from `createPicker`. Once you do this, you should be able to get rid of the class declaration and keep the only the function. Then inside the control menu, you will have:

```
internal fun ArtMakerControlMenu(
    state: ArtMakerUIState,
    onAction: (ArtMakerAction) -> Unit,
    onDrawEvent: (DrawEvent) -> Unit,
    modifier: Modifier = Modifier,
    onShowStrokeWidthPopup: () -> Unit,
    setBackgroundImage: (ImageBitmap?) -> Unit,
    imageBitmap: ImageBitmap?,
    artMakerConfiguration: ArtMakerConfiguration,
    onActivateEraser: () -> Unit,
    isEraserActive: Boolean,
//    imagePicker: ImagePicker
) {
    val imagePicker = createPicker()
```
This will keep the main `Artmaker` composable as is and still do the job. There will be less work to initialise anything inside the calling app's main activity class.

Reply:
Good idea